core/hash_map.h

```diff
@@ -29,12 +29,64 @@
 #ifndef HASH_MAP_H
 #define HASH_MAP_H
 
+#include <cmath>
+
 #include "hashfuncs.h"
 #include "error_macros.h"
 #include "ustring.h"
 #include "os/memory.h"
 #include "list.h"
 
+// Don't pollute global namespace with generic terms like 'compare' or 'enable_if'
+namespace _hashmap_priv {
```

Comment:
If it is preferred I can call it _hash_compare or something. I just figured this was easier to read especially since we really don't expect other callers for this. I was under the impression that @reduz didn't want namespaces to separate parts of the engine, but that he was fine with using it for this type of hiding. Maybe I misunderstood.